src/pygeoprocessing/geoprocessing_core.pyx

```diff
@@ -941,6 +942,38 @@ def _raster_band_percentile_double(
             # you never know if this might fail!
             LOGGER.warning('unable to remove %s', file_path)
     if rm_dir_when_done:
-        shutil.rmtree(working_sort_directory)
+        _safe_rmtree(working_sort_directory)
     LOGGER.debug('here is percentile_list: %s', str(result_list))
     return result_list
+
+
+def _safe_rmtree(path):
+    """Attempts to remove a file path and logs a warning if it fails.
+
+    There is a possible bug in shutil.rmtree that can raise a FileNotFound
+    error even if the removal of a directory is possible and it removed it
+    see: https://bugs.python.org/issue29699. This function is effectively a
+    guard against that bug and logs a warning if the file path could not
+    be removed.
+
+    Args:
+        path (str): represents a file path to remove
```

Comment:
`shutil.rmtree` expects `path` to be a directory. The argument description should reflect that OR if we want this to be a _general_ remove, we should probably call it something other than `rmtree`.

Reply:
I changed the docstring to reflect directory paths in d309f81cad51f8f152ca4e80ec8a6d79161c32b0